File: generate-three-videos.mjs
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { spawn } from 'child_process';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// 3つのトピック用のビデオ生成
const videosToGenerate = [
    {
        topic: "劇的な逆転勝利",
        comment: "あー今9回2死でダメかと思ったわ、やりやがった",
        number: 1
    },
    {
        topic: "守護神のセーブ失敗",
        comment: "守護神がまさかのセーブ失敗で大荒れや",
        number: 2
    },
    {
        topic: "新人選手の活躍",
        comment: "新人くんが決勝ホームラン打ってくれた！",
        number: 3
    }
];

async function generateSingleVideo(videoInfo) {
    const outputDir = path.join(__dirname, 'output', 'three-videos');

    // 出力ディレクトリを作成
    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir, { recursive: true });
    }

    const safeTopic = videoInfo.topic.replace(/の|から|選手/g, '').substring(0, 10);
    const outputPath = path.join(outputDir, `video_${videoInfo.number.toString().padStart(3, '0')}_${safeTopic}.mp4`);

    console.log(`ビデオ ${videoInfo.number}: ${videoInfo.topic} を生成中...`);

    return new Promise((resolve, reject) => {
        const pythonProcess = spawn('python', [
            path.join(__dirname, 'python', 'video_composer.py'),
            videoInfo.comment,
            outputPath
        ], {
            cwd: __dirname,
            stdio: ['pipe', 'pipe', 'pipe']
        });

        let stdout = '';
        let stderr = '';

        pythonProcess.stdout.on('data', (data) => {
            stdout += data.toString();
        });

        pythonProcess.stderr.on('data', (data) => {
            stderr += data.toString();
        });

        pythonProcess.on('close', (code) => {
            if (code === 0) {
                console.log(`OK ビデオ ${videoInfo.number} 生成成功`);
                resolve(true);
            } else {
                console.log(`NG ビデオ ${videoInfo.number} 生成失敗 (code: ${code})`);
                console.log('stdout:', stdout);
                console.log('stderr:', stderr);
                resolve(false);
            }
        });

        pythonProcess.on('error', (error) => {
            console.log(`NG ビデオ ${videoInfo.number} エラー:`, error.message);
            resolve(false);
        });
    });
}

async function main() {
    console.log('=== 3つのビデオ生成開始 ===');

    let successCount = 0;

    for (const videoInfo of videosToGenerate) {
        const success = await generateSingleVideo(videoInfo);
        if (success) {
            successCount++;
        }
    }

    console.log('\n=== 生成完了 ===');
    console.log(`成功: ${successCount}/3`);
    console.log(`成功率: ${(successCount/3*100).toFixed(1)}%`);
}

main().catch(console.error);

File: generate-three-videos.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import json
import os
import subprocess
import sys
from datetime import datetime

def load_analysis_data():
    """分析データを読み込み"""
    data_path = r"C:\Users\pukur\nanj-baseball\output\nanj-sites-2025-09-13\nanj-analysis-result.json"
    with open(data_path, 'r', encoding='utf-8') as f:
        return json.load(f)

def generate_single_video(topic_name, comment_text, video_number):
    """1つのビデオを生成"""
    print(f"ビデオ {video_number}: {topic_name} を生成中...")

    output_dir = r"C:\Users\pukur\nanj-baseball\output\three-videos"
    os.makedirs(output_dir, exist_ok=True)

    # ファイル名を短縮
    safe_topic = topic_name.replace("の", "").replace("から", "").replace("選手", "")[:10]
    output_path = os.path.join(output_dir, f"video_{video_number:03d}_{safe_topic}.mp4")

    # Pythonスクリプトを呼び出し
    cmd = [
        sys.executable,
        r"C:\Users\pukur\nanj-baseball\python\video_composer.py",
        comment_text,
        output_path
    ]

    try:
        result = subprocess.run(cmd, capture_output=True, text=True, encoding='utf-8')
        if result.returncode == 0:
            print(f"OK ビデオ {video_number} 生成成功: {os.path.basename(output_path)}")
            return True
        else:
            print(f"NG ビデオ {video_number} 生成失敗: {result.stderr}")
            return False
    except Exception as e:
        print(f"NG ビデオ {video_number} エラー: {str(e)}")
        return False

def main():
    print("=== 3つのビデオ生成開始 ===")

    # 分析データを読み込み
    data = load_analysis_data()

    # 各トピックから1つずつコメントを選択
    videos_to_generate = [
        {
            "topic": "劇的な逆転勝利",
            "comment": "あー今9回2死でダメかと思ったわ、やりやがった",
            "number": 1
        },
        {
            "topic": "守護神のセーブ失敗",
            "comment": "守護神がまさかのセーブ失敗で大荒れや",
            "number": 2
        },
        {
            "topic": "新人選手の活躍",
            "comment": "新人くんが決勝ホームラン打ってくれた！",
            "number": 3
        }
    ]

    success_count = 0

    for video_info in videos_to_generate:
        success = generate_single_video(
            video_info["topic"],
            video_info["comment"],
            video_info["number"]
        )
        if success:
            success_count += 1

    print(f"\n=== 生成完了 ===")
    print(f"成功: {success_count}/3")
    print(f"成功率: {success_count/3*100:.1f}%")

if __name__ == "__main__":
    main()